SingularityUI/app/components/requests/RequestFilters.jsx

```diff
@@ -78,6 +78,16 @@ export default class RequestFilters extends React.Component {
     }
   ];
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filter: _.extend({}, this.props.filter),
+    };
+
+    this.debouncedOnFilterChange = _.debounce(this.props.onFilterChange, 250);
+  }
+
   handleStatusSelect(selectedKey) {
     this.props.onFilterChange(_.extend({}, this.props.filter, {state: RequestFilters.REQUEST_STATES[selectedKey].filterVal}));
```

Comment:
left the other filter handlers as is because they're not causing any issues, though it does make the code a bit inconsistent